source/server/hot_restarting_base.cc

```diff
@@ -36,8 +37,11 @@ sockaddr_un HotRestartingBase::createDomainSocketAddress(uint64_t id, const std:
   initDomainSocketAddress(&address);
   Network::Address::PipeInstance addr(fmt::format(socket_path + "_{}_{}", role, base_id_ + id),
                                       socket_mode, nullptr);
-  memcpy(&address, addr.sockAddr(), addr.sockAddrLen());
+  MemBlockBuilder<sockaddr> mem_builder(addr.sockAddrLen());
+  mem_builder.appendData(absl::Span<const sockaddr>(addr.sockAddr(), addr.sockAddrLen()));
+  SAFE_MEMCPY(&address, mem_builder.release().get());
```

Comment:
The new code has an extra data-copy compared to the old code. In this case I doubt it matters that much as I don't think hot-restart is a frequent operation. However, I want to make sure we establish precedent of optimal-but-safe use.

In this case I think the old code is actually dangerous because there's nothing there to guarantee that sockAddrLen() is smaller than sizeof(address). 

I think the new code is going to allocate a whole array of sockaddr as a temporary buffer. Really you just need one but we need to know it's the right size.

I think the right approach is to move the SAFE_MEMCPY into the implementation of PipeInstance, so that the sizeof is run in contexts where the compiler can know the right sizes. WDYT?

Reply:
Sure, so I agree on what you say about the extra copy, actually I had plans to see whether I could avoid SAFE_MEMCPY for this. 

However, it is not very clear why using SAFE_MEMCPY in PipeInstance is necessary, maybe a method that returns o transform it into a sockaddr_un (like address) is enough.

Reply:
Agreed; if we can do this transfer without memcpy that seems reasonable.